core/src/main/java/hudson/PluginManager.java

```diff
@@ -1916,7 +1917,11 @@ public HttpResponse doInstallNecessaryPlugins(StaplerRequest req) throws IOExcep
     public Map<String,VersionNumber> parseRequestedPlugins(InputStream configXml) throws IOException {
         final Map<String,VersionNumber> requestedPlugins = new TreeMap<>();
         try {
-            SAXParserFactory.newInstance().newSAXParser().parse(configXml, new DefaultHandler() {
+            SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+            // this is no real issue, it is just a false positive, since jenkins handles it differently
+            // for more information check: https://github.com/jenkinsci/jenkins/pull/4779
+            parserFactory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
```

Comment:
Really ought to use the feature "http://apache.org/xml/features/disallow-doctype-decl". See the findsecbugs or OWASP cheat sheet pages. The "disallow" feature is more thorough, to prohibit XXE. The secure processing feature only gets most of the way there.

Reply:
@jeffret-b I understand your explanation, but the link is pointing to a 404 page. Could you refer to the correct one?

Reply:
It's confusing, but that isn't really a link. It's more of a URI than a URL, with no actual target at that location.

Here's are a couple of actual links that talk about using this identifier: [OWASP Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html) and [findsecbus](SAXParserFactory). The findsecbugs page lists the `FEATURE_SECURE_PROCESSING` as the first solution and the `disallow-doctype-decl` second, though it should present them in the other order. The second is more thorough, as described in the OWASP document. Either one quiets findsecbugs, probably because they figure if you've added one you've thought about the problem and the level of protection you need. Given how complex this is, I don't think that's a valid assumption.